### PR TITLE
[FLINK-34429] [flink-kubernetes] Adding K8S Annotations to Internal Service

### PR DESCRIPTION
**Pull request by barakbn**
## What is the purpose of the change

Flink currently supports adding Annotations to the Rest Service (with the configuration key:
kubernetes.rest-service.annotations). 

This PR adds a new configuration key: kubernetes.internal-service.annotations  to allow the same functionality for the internal service. 

## Brief change log

*(for example:)*
  - New configuration key: "kubernetes.internal-service.annotations"  is added
  - Configuration value is used in intenal-service creation
  - Documentation is added

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no 
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: yes
  - The S3 file system connector: no

## Documentation

  - Does this pull request introduce a new feature? yes
  - If yes, how is the feature documented? docs

### CODE REVIEW

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/parameters/KubernetesJobManagerParameters.java [hunk 1]
```diff
@@ -118,6 +118,12 @@ public Map<String, String> getRestServiceAnnotations() {
                 .orElse(Collections.emptyMap());
     }
 
+    public Map<String, String> getInternalServiceAnnotations() {
+        return flinkConfig
+                .getOptional(KubernetesConfigOptions.REST_SERVICE_ANNOTATIONS)
+                .orElse(Collections.emptyMap());
+    }
+
```
Comment (dannycranmer) [inline flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/parameters/KubernetesJobManagerParameters.java:126]: Can you please add a test to `KubernetesJobManagerParametersTest`

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/services/HeadlessClusterIPService.java [hunk 2]
```diff
@@ -50,6 +50,7 @@ public Service buildUpInternalService(
                 .withNewMetadata()
                 .withName(serviceName)
                 .withLabels(kubernetesJobManagerParameters.getCommonLabels())
+                .withAnnotations(kubernetesJobManagerParameters.getInternalServiceAnnotations())
```
Comment (dannycranmer) [inline flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/services/HeadlessClusterIPService.java:53]: This change is not tested. I cannot see an existing test for this method, and am not sure why that is missing. Can we add one here: `InternalServiceDecoratorTest` ?

**Review by dannycranmer — CHANGES_REQUESTED** [review]

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/services/HeadlessClusterIPService.java (hunk 2 above)
Comment (barakbn) [reply flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/services/HeadlessClusterIPService.java:53]: Done.

File: flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/parameters/KubernetesJobManagerParameters.java (hunk 1 above)
Comment (barakbn) [reply flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/parameters/KubernetesJobManagerParameters.java]: Done.

**Review by dannycranmer — APPROVED** [review]
LGTM, thanks @barakbn

**Comment by dannycranmer** [comment]
@barakbn please rebase and squash commits and I will merge.

**Pull request merged**